addons/popochiu/editor/main_dock/popochiu_row/object_row/room_row/popochiu_room_row.gd

```diff
@@ -0,0 +1,86 @@
+@tool
+extends "res://addons/popochiu/editor/main_dock/popochiu_row/object_row/popochiu_object_row.gd"
+
+enum RoomOptions {
+	DELETE = MenuOptions.DELETE,
+	ADD_TO_CORE = Options.ADD_TO_CORE,
+	CREATE_STATE_SCRIPT = Options.CREATE_STATE_SCRIPT,
+	SET_AS_MAIN,
+}
+
+const STATE_TEMPLATE = "res://addons/popochiu/engine/templates/room_state_template.gd"
+
+var is_main := false : set = set_is_main
+
+@onready var btn_play: Button = %BtnPlay
+
+
+#region Godot ######################################################################################
+func _ready() -> void:
+	super()
+	
+	# Assign icons
+	tag.texture = get_theme_icon("Heart", "EditorIcons")
+	btn_play.icon = get_theme_icon("MainPlay", "EditorIcons")
+	
+	menu_popup.set_item_disabled(menu_popup.get_item_index(Options.ADD_TO_CORE), true)
+	menu_popup.set_item_disabled(menu_popup.get_item_index(Options.CREATE_STATE_SCRIPT), true)
+	
+	btn_play.pressed.connect(_play)
+
+
+#endregion
+
+#region Virtual ####################################################################################
+func _get_state_template() -> Script:
+	return load(STATE_TEMPLATE)
+
+
+#endregion
+
+#region SetGet #####################################################################################
+func set_is_main(value: bool) -> void:
+	is_main = value
+	tag.visible = value
+	
+	menu_popup.set_item_disabled(menu_popup.get_item_index(RoomOptions.SET_AS_MAIN), value)
+
+
+#endregion
+
+#region Private ####################################################################################
+func _get_menu_cfg() -> Array:
+	return [
+		{
+			id = RoomOptions.SET_AS_MAIN,
+			icon = get_theme_icon("Heart", "EditorIcons"),
+			label = "Set as Main scene",
+		},
+	] + super()
+
+
+func _menu_item_pressed(id: int) -> void:
+	match id:
+		RoomOptions.SET_AS_MAIN:
+			PopochiuEditorHelper.signal_bus.main_scene_changed.emit(path)
```

Comment:
Same as for player character, shouldn't this be emitted in the setter?

Reply:
I tried that, but it caused some issues I can't remember now. I guess it was related with the initial value of the property. But I can give it a second try.

Reply:
Done.